File: scripts/video_feed.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String
from sensor_msgs.msg import Image
from cv_bridge import CvBridge, CvBridgeError
import cv2
import numpy as np


rospy.init_node("camera",anonymous = True)
pub = rospy.Publisher("video_feed", Image, queue_size = 1)
rate = rospy.Rate(10)
bridge = CvBridge()
cap = cv2.VideoCapture(0)
while not rospy.is_shutdown():
    ret, frame = cap.read()
    gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
    messg = bridge.cv2_to_imgmsg(frame, "bgr8")
    pub.publish(messg)
    cv2.imshow("webcam", frame)
    cv2.waitKey(100)
    rate.sleep()

File: src/video_feed.cpp
#include "opencv2/highgui/highgui.hpp"
#include "opencv2/core/core.hpp"
#include "opencv2/imgproc/imgproc.hpp"
#include "ros/ros.h"
#include<bits/stdc++.h>
#include<image_transport/image_transport.h>
#include<cv_bridge/cv_bridge.h>


using namespace std;
using namespace cv;
using namespace ros;


int main(int argc, char **argv){
    init(argc, argv, "camera");
    NodeHandle n;
    VideoCapture cap(0);
    if(!cap.isOpened())
	    return -1;
    image_transport::ImageTransport it(n);
    image_transport::Publisher video_pub = it.advertise("videofeed", 1);
    Rate loop_rate(10);
    while(n.ok()){
	    Mat img;
	    cap>>img;
        sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), "bgr8", img).toImageMsg();
        video_pub.publish(msg);
        namedWindow("WebCam", WINDOW_NORMAL);
	    imshow("WebCam",img);
    	waitKey(100);
        spinOnce();
        loop_rate.sleep();
    }
    return 0;
}